azure/scope/cluster.go

```diff
@@ -253,7 +253,7 @@ func (s *ClusterScope) LBSpecs() []azure.ResourceSpecGetter {
 	// Node outbound LB
 	if s.NodeOutboundLB() != nil {
 		specs = append(specs, &loadbalancers.LBSpec{
-			Name:                 s.NodeOutboundLBName(),
+			Name:                 s.NodeOutboundLB().Name,
```

Comment:
We should either change the `NodeOutboundLBName` `ClusterScope` method or get rid of it. Because the method is only called here it's probably easiest to delete it.